x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryResponse.java

```diff
@@ -396,41 +390,15 @@ public EsqlResponse responseInternal() {
         return esqlResponse;
     }
 
-    public static class Profile implements Writeable {
-        private final List<DriverProfile> drivers;
-
-        public Profile(List<DriverProfile> drivers) {
-            this.drivers = drivers;
-        }
+    public record Profile(List<DriverProfile> drivers) implements Writeable {
```

Comment:
I wonder if we even need that? May be it is worth in-lining List<DriverProfile> instead?